pyccl/tests/test_tk3d.py

```diff
@@ -137,3 +137,38 @@ def test_tk3d_eval(is_product):
     ptrue = tkkaf(ktest[None, :], ktest[:, None], atest)
     phere = tsp.eval(ktest, atest)
     assert_allclose(phere.flatten(), ptrue.flatten(), rtol=1E-6)
+
+
+@pytest.mark.parametrize('is_product', [True, False])
+def test_tk3d_spline_arrays(is_product):
+    (a_arr, lk_arr, fka1_arr, fka2_arr, tkka_arr) = get_arrays()
+    if is_product:
+        tsp = ccl.Tk3D(a_arr, lk_arr, pk1_arr=fka1_arr, pk2_arr=fka2_arr)
+    else:
+        tsp = ccl.Tk3D(a_arr, lk_arr, tkk_arr=tkka_arr)
+
+    a_get, lk_get1, lk_get2, out = tsp.get_spline_arrays()
+    assert np.allclose(a_get, a_arr, rtol=1e-15)
+    assert np.allclose(lk_get1, lk_arr, rtol=1e-15)
+    assert np.allclose(lk_get2, lk_arr, rtol=1e-15)
+
+    if is_product:
+        assert np.allclose(np.log(out[0]), fka1_arr, rtol=1e-15)
+        assert np.allclose(np.log(out[1]), fka2_arr, rtol=1e-15)
+    else:
+        assert np.allclose(np.log(out[0]), tkka_arr, rtol=1e-15)
+
+
+def test_tk3d_spline_arrays_raises():
+    (a_arr, lk_arr, fka1_arr, fka2_arr, tkka_arr) = get_arrays()
+    tsp = ccl.Tk3D(a_arr, lk_arr, tkk_arr=tkka_arr)
+
+    if hasattr(tsp.__class__, "has_tsp"):
+        # `has_tsp` is a property
+        delattr(tsp, "tsp")
```

Comment:
Does this cause memory leaks?

Reply:
Maybe add a comment, just in case this gets flagged by some checks couple years down the road.

Reply:
Okay, I tweaked that bit just in case PR923 doesn't make the cut, and I also added a line to free the C-memory.